src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -4463,186 +4463,252 @@ bool Pusher::overlapsWithIOBlockage(const odb::Rect& cluster_box,
 
 //////// Snapper ////////
 
-Snapper::Snapper() : inst_(nullptr)
+Snapper::Snapper(utl::Logger* logger) : logger_(logger), inst_(nullptr)
 {
 }
 
-Snapper::Snapper(odb::dbInst* inst) : inst_(inst)
+Snapper::Snapper(utl::Logger* logger, odb::dbInst* inst)
+    : logger_(logger), inst_(inst)
 {
 }
 
 void Snapper::snapMacro()
 {
-  const odb::Point snap_origin = computeSnapOrigin();
-  inst_->setOrigin(snap_origin.x(), snap_origin.y());
+  snap(true /*horizontal snap*/);
+  snap(false /*horizontal snap*/);
```

Comment:
Shouldn't it be `vertical snap`?

Reply:
Afaiu, we snap horizontally to align the pins of a layer whose preferred direction is vertical:

https://github.com/The-OpenROAD-Project/OpenROAD/blob/934fad8a8cf47f56cccd918da3ef0160447a4433/src/mpl2/src/hier_rtlmp.cpp#L4533-L4539

Reply:
Oh, I see. My intentions were to replicate the boolean name, but apparently it didn't make it very clear. I'll write exaclty what each one is doing:
```
  snap(true); // Horizontal snap
  snap(false); // Vertical snap